Adekanmi Israel/faisal-code-composition.ts

```diff
@@ -0,0 +1,126 @@
+// laptop.ts
+class Laptop {
+    keyboards: Keyboard[] = [];
+    nic: NetworkInterfaceCard;
+    screen: Display;
+    hardDisk: HardDisk;
+    powerState: boolean = false;
+    bitKind: BitKind;
+    os: OS;
+    allUpdateLogs: string[] = [];
+  
+    constructor(
+      keyboard: Keyboard,
+      bitKind: BitKind,
+      display: Display,
+      nic: NetworkInterfaceCard,
+      os: OS,
+      hardDisk: HardDisk
+    ) {
+      if (!keyboard ||!bitKind ||!display ||!nic ||!os ||!hardDisk) {
+        throw new Error("All parameters are required");

```

Comment:
Awesome, you're the only person that put validation.
You can share it with others.